File: app/src/main/java/com/cn29/aac/ui/shopping/vm/ShoppingActivityViewModel.java
package com.cn29.aac.ui.shopping.vm;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

/**
 * Created by Charles Ng on 23/10/2017.
 */

public class ShoppingActivityViewModel extends ViewModel {

  private MutableLiveData<String> queryData;

  public ShoppingActivityViewModel() {
    this.queryData = new MutableLiveData<>();
  }

  public void query(String query) {
    queryData.setValue(query);
  }

  public LiveData<String> getQueryData() {
    return queryData;
  }
}

File: app/src/main/java/com/cn29/aac/ui/shopping/vm/ShoppingActivityViewModel.kt
package com.cn29.aac.ui.shopping.vm

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * Created by Charles Ng on 23/10/2017.
 */
class ShoppingActivityViewModel : ViewModel() {
    private val queryData: MutableLiveData<String> = MutableLiveData()
    fun query(query: String) {
        queryData.value = query
    }

    fun getQueryData(): LiveData<String> {
        return queryData
    }

}